project/apps/web/stores/useSocketStore.ts

```diff
@@ -0,0 +1,55 @@
+import { io, Socket } from "socket.io-client";
+import { create } from "zustand";
+
+const SOCKET_SERVER_URL = "http://localhost:8080";
+
+interface SocketState {
+  socket: Socket | null;
+  isConnected: boolean;
+  connect: () => void;
+  disconnect: () => void;
+}
+
+const useSocketStore = create<SocketState>((set, get) => ({
+  socket: null,
+  isConnected: false,
+  connect: () => {
+    if (get().socket) return; // Prevent multiple connections
+
+    const socket = io(SOCKET_SERVER_URL, {
+      reconnectionAttempts: 5,
+      withCredentials: true,
+    });
+
+    socket.on("connect", () => {
+      set({ isConnected: true });
+      console.log("Connected to Socket.IO server");
+    });
+
+    socket.on("disconnect", () => {
+      set({ isConnected: false });
+      console.log("Disconnected from Socket.IO server");
+    });
+
```

Comment:
not sure about reconnect_failed cause we have auto retry mechanism but i added for `connect_error`, but we can add in the future if we see the need for it since im leaving the implementation out of this version currently